### PR TITLE
Remove incorrect deprecation warning when using useMutation

### PR DESCRIPTION
**Pull request by aericson**
Closes: #1297 

### What are the changes and their implications?
See #1297. There is a warning, of deprecation, telling to use `invoke()` when one is using `useMutation`.
I've also added a simple test for the useMutation since there wasn't any.

### Checklist

- [x] Tests added for changes
- ~[ ] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes~

### CODE REVIEW

File: packages/core/src/use-mutation.ts [hunk 1]
```diff
@@ -15,10 +15,13 @@ export function useMutation<TResult, TError = unknown, TVariables = undefined, T
...
 ) {
-  validateQueryFn(mutationResolver)
+  const enhancedResolverRpcClient = sanitize(mutationResolver)
 
-  return useReactQueryMutation(mutationResolver, {
-    throwOnError: true,
-    ...config,
-  }) as MutationResultPair<TResult, TError, TVariables, TSnapshot>
+  return useReactQueryMutation(
+    (variables?: TVariables) => enhancedResolverRpcClient(variables, {fromQueryHook: true}),
```
Comment (flybayer) [inline packages/core/src/use-mutation.ts:21]: Unless I'm mistaking, the previous type for this was `MutationFunction` which is typed `(variables: TVariables)` and so the new code should be like this to match.
```suggestion
    (variables: TVariables) => enhancedResolverRpcClient(variables, {fromQueryHook: true}),
```

**Review by flybayer — COMMENTED** [review]
Sweet, looks great! Just one comment.

**Comment by aericson** [comment]
> Sweet, looks great! Just one comment.

good eye! thanks, changed it.

**Review by flybayer — APPROVED** [review]
Awesome!

**Pull request merged**